sql_extensions/migrations/special-queries.sql

```diff
@@ -47,15 +47,54 @@ WHERE
 ORDER BY
     (data -> 'market_metadata' ->> 'market_id')::NUMERIC,
     (data ->> 'transaction_version')::NUMERIC DESC,
-    (data ->> 'event_index')::NUMERIC DESC;
+    (data -> 'state_metadata' ->> 'market_nonce')::NUMERIC DESC;
 
 CREATE OR REPLACE FUNCTION UPDATE_LATEST_STATE()
 RETURNS TRIGGER AS $$
 BEGIN
-  DELETE FROM inbox_latest_state
-  WHERE (inbox_latest_state.data->'market_metadata'->>'market_id')::numeric = (NEW.data->'market_metadata'->>'market_id')::numeric
-  AND (inbox_latest_state.data->'state_metadata'->>'bump_time')::numeric <= (NEW.data->'state_metadata'->>'bump_time')::numeric;
-  INSERT INTO inbox_latest_state SELECT NEW.*;
+  IF EXISTS (
+    SELECT 1
+    FROM inbox_latest_state
+    WHERE (inbox_latest_state.data->'market_metadata'->>'market_id')::numeric = (NEW.data->'market_metadata'->>'market_id')::numeric
+      AND (inbox_latest_state.data->'state_metadata'->>'market_nonce')::numeric < (NEW.data->'state_metadata'->>'market_nonce')::numeric
+  ) THEN
+    RETURN NEW;
+  END IF;
```

Comment:
If I'm not mistaking, this will never work. Did you mean

```suggestion
  IF EXISTS (
    SELECT 1
    FROM inbox_latest_state
    WHERE (inbox_latest_state.data->'market_metadata'->>'market_id')::numeric = (NEW.data->'market_metadata'->>'market_id')::numeric
      AND (inbox_latest_state.data->'state_metadata'->>'market_nonce')::numeric > (NEW.data->'state_metadata'->>'market_nonce')::numeric
  ) THEN
    RETURN NEW;
  END IF;
```

Reply:
Yes that's my bad, meant to change that. That would explain it not updating lol

I think that, in addition to another follow-up issue (the `market_data` isn't being created for markets until the first 1 minute period ends) covers most of the "why isn't this showing up?" errors